File: Atividade08/java/composicao.java
public class Poupanca{
  float saldo;
  void saque(){
    System.out.println("Novo saldo...");
  }
  void deposito(){
    System.out.println("Novo saldo...");
  }
}

public class ContaCorrente{
  float saldo;
  void saque(){
    System.out.println("Novo saldo...");
  }
  void deposito(){
    System.out.println("Novo saldo...");
  }
}

public class Banco{
  Poupanca[] pops;
  ContaCorrente[] cc;
  int numConta, numPoupanca;
  void iniciaBanco(){
    pops = new Poupanca[100];
    cc = new ContaCorrente[100];
    numConta = 1;
    numPoupanca = 1;
  }
  void falencia(){
    for(int i = 0; i < 100; i++){
      pops[i] = null;
      cc[i] = null;
    }
  }
}

File: Atividade09/code/composicao.js
class Poupanca {
    constructor(saldo) {
        this.saldo = saldo;
    }
    saque() {
        this.saldo -= 10.0
        console.log(`Novo saldo ${this.saldo}`);
    }
    deposito() {
        this.saldo += 10.0
        console.log(`Novo saldo ${this.saldo}`);
    }
}
class ContaCorrente {
    constructor(saldo) {
        this.saldo = saldo;
    }
    saque() {
        this.saldo -= 100.0
        console.log(`Novo saldo ${this.saldo}`);
    }
    deposito() {
        this.saldo += 100.0
        console.log(`Novo saldo ${this.saldo}`);
    }
}
class Banco {
    constructor() {
        this.pops = [];
        this.cc = [];
    }
    abreContaCorrente() {
        this.cc.push(new ContaCorrente(0.0));
    }
    abrePoupanca() {
        this.pops.push(new Poupanca(300.0));
    }
    falencia() {
        this.pops.splice(0,this.pops.length);
        this.cc.splice(0,this.cc.length);
        this.pops = null;
        this.cc = null;
    }
}

